app/javascript/App/bundles/CoinShow/HistoricalPriceDataTableWrapper.tsx

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react'
+import HistoricalPriceDataTable from './HistoricalPriceDataTable'
+import CurrencyContext, {
+  CurrencyContextType,
+} from '~/bundles/common/contexts/CurrencyContext'
+
+const HistoricalPriceDataTableWrapper = (props) => (
+  <CurrencyContext.Consumer>
```

Comment:
This Consumer seems to be used quite a bit. I wonder if its worth creating a separate HOC for it to feed the props into the wrapped component. Similar to `withStyles()`. It would save having to create a new component like this.